data/src/test/java/io/gnosis/data/repositories/TransactionRepositoryTest.kt

```diff
@@ -0,0 +1,391 @@
+                assertEquals(transactionDto.blockTimestamp, date)
+            }
+        }
+
+    @Test
+    fun `dataSizeBytes (one byte data) should return 1`() {
+        assertEquals("0x0A".dataSizeBytes(), 1L)
+    }
+
+    @Test
+    fun `dataSizeBytes (zero byte data) should return 0`() {
+        assertEquals("0x".dataSizeBytes(), 0L)
+    }
+
+    @Test
+    fun `dataSizeBytes (5 byte data) should return 5`() {
+        assertEquals("0x0123456789".dataSizeBytes(), 5L)
+    }
+
+    @Test
+    fun `hexStringNullOrEmpty (is 0x) should return true`() {
+        assertTrue("0x".hexStringNullOrEmpty())
+    }
+
+    @Test
+    fun `hexStringNullOrEmpty (is null) should return true`() {
+        assertTrue(null.hexStringNullOrEmpty())
+    }
+
+    @Test
+    fun `hexStringNullOrEmpty (is 0x1A) should return false`() {
+        assertFalse("0x1A".hexStringNullOrEmpty())
+    }
+
+    private fun buildModuleTransactionDto(
+        to: Solidity.Address = defaultToAddress,
+        module: Solidity.Address,
+        safe: Solidity.Address = defaultSafeAddress
+    ): ModuleTransactionDto =
+        ModuleTransactionDto(
+            to = to,
+            module = module,
+            safe = safe,
+            value = BigInteger.ZERO
+        )
+
+    private fun buildMultisigTransactionDto(
+        transfers: List<TransferDto>? = null,
+        contractInfoType: ContractInfoType? = null,
+        dataDecodedDto: DataDecodedDto? = null,
+        safe: Solidity.Address = defaultSafeAddress,
+        to: Solidity.Address = defaultToAddress,
+        operation: Operation = Operation.CALL
+    ): MultisigTransactionDto {
+        return MultisigTransactionDto(
+            safe = safe,
+            to = to,
+            operation = operation,
+            value = BigInteger.ONE,
+            nonce = BigInteger.ONE,
+            safeTxGas = BigInteger.ONE,
+            baseGas = BigInteger.ONE,
+            gasPrice = BigInteger.ONE,
+            transfers = transfers,
+            contractInfo = contractInfoType?.let { ContractInfoDto(it) },
+            dataDecoded = dataDecodedDto
+        )
+    }
+
+    private fun buildEthereumTransactionDto(
+        transfers: List<TransferDto> = listOf(),
+        from: Solidity.Address = defaultFromAddress,
+        to: Solidity.Address = defaultToAddress,
+        data: String? = "0x",
+        value: BigInteger = BigInteger.ONE
+    ): EthereumTransactionDto {
+        return EthereumTransactionDto(
+            from = from,
+            to = to,
+            value = value,
+            transfers = transfers,
+            data = data,
+            txHash = "0x1234",
+            blockTimestamp = null
+        )
+    }
+
+    private fun buildTransferDto(
+        type: TransferType = TransferType.ETHER_TRANSFER
+    ): TransferDto =
+        TransferDto(
+            to = defaultToAddress,
+            from = defaultFromAddress,
+            type = type,
+            value = BigInteger.ONE
+        )
+}
```

Comment:
Generally, look really good, man. I am just missing some `coVerify { transactionApi.loadTransactions(address) }` calls. Although it's only one method under the hood for now but it could change in the future.

Reply:
I added those